config/fields/number.php

```diff
@@ -36,7 +36,7 @@
                 return null;
             }
 
-            return (float)Str::float($value);
+            return Str::float($value);
```

Comment:
Just thinking myself... Maybe we should rename this methods to `toValue` and add another `toNumber` one that just does `return (float)$this->toValue($value)`, @bastianallgeier?